draft-ietf-quic-transport.md

```diff
@@ -1641,6 +1641,13 @@ of connection establishment.  By giving the client a different connection ID to
 use, a server can cause the connection to be routed to a server instance with
 more resources available for new connections.
 
+If a server receives a client Initial with an invalid Retry token,
+it knows the client will not accept another Retry token.  It can either
```

Comment:
I don't understand this sentence. How would the server know this for sure?

Reply:
That's a keen observation (I think).

There is a chance where a Retry token is considered invalid due to packet becoming corrupt while being transmitted. A server can only say for sure that "the client will not accept another Retry token" unless either of the following conditions are met:
* The token is accompanied by a checksum (or an integrity protection). The checksum is correct, but the contents of the Retry token is invalid.
* The server successfully unprotects the Initial.

Reply:
The client could also connect to an unexpected endpoint or the token could have expired.

Reply:
Let me retract my [comment above](https://github.com/quicwg/base-drafts/pull/3107/files#r341481158). We do not need to care about the packet becoming corrupt on the wire, because the prerequisite for responding with a CONNECTION_CLOSE is that the Initial that has been received can be unprotected.

To avoid confusion, maybe change "If a server receives a client Initial with an invalid Retry token" to "If a server receives a client Initial that can be unprotected but contains an invalid Retry token"?

> The client could also connect to an unexpected endpoint or the token could have expired.

I'd say that that's a failure in the server deployment. We do not discuss how that should be handled.

Reply:
> I'd say that that's a failure in the server deployment. We do not discuss how that should be handled.

Since Retry is processed before handshake, why can't it just be the client sending to the wrong IP address?

Reply:
#3128 requires being able to distinguish between the two types of tokens.  I updated the text with Kazuho's suggestion about being able to unprotect the Initial.